port/raspberrypi/rp2xxx/src/hal/drivers.zig

```diff
@@ -113,6 +122,23 @@ pub const I2C_Device = struct {
             error.NoData => return 0,
         };
     }
+
+    fn writev_then_readv_fn(dd: *anyopaque, write_chunks: []const []const u8, read_chunks: []const []u8) (WriteError || ReadError)!void {
+        const dev: *I2C_Device = @ptrCast(@alignCast(dd));
+        return dev.write_then_readv(write_chunks, read_chunks) catch |err| switch (err) {
```

Comment:
writev_then_readv.

make sure to try to build your driver with these changes, because zig's lazy compilation will miss these issues.

Reply:
Good idea.  zig build shows that I do have to update all the boards after all.  I'll plug away at them as time allows.

Reply:
Actually, some of the boards, like espressif/esp, don't have write_then_read at all.